openedx/openedx-platform
# Run modulestore tests in common/lib/... using Django

## Description

This PR changes the modulestore tests to always run with Django enabled, using settings similar to the LMS.

Why? Currently parts of the modulestore test suite are designed to run "without django", although there is still a lot of django functionality imported at times, and some of the tests do in fact use django. But for my other PR https://github.com/edx/edx-platform/pull/27565 (moving split's course indexes from MongoDB to MySQL), we will need to always have Django enabled. So this PR paves the way for that one.

The previous tests that did use Django used a special settings file, [`openedx/tests/settings.py`](https://github.com/edx/edx-platform/blob/61b9a523e74433260f4d0e22204910fd3ab28b0c/openedx/tests/settings.py) which made some debugging confusing because those tests had quite different django settings than other tests. This change deletes that file and runs the tests using [the LMS `test` settings](https://github.com/edx/edx-platform/blob/master/lms/envs/test.py). (TBD: would it be better to use the CMS settings? Either way is an option.)

The test suite also contains many different ways of initializing and testing a modulestore, with significant differences in their configuration, and also a lot of repetition. I find this makes understanding, debugging and writing tests more difficult. So this PR also reduces the number of different "test case using modulestore" base classes:
* Simplifies `MixedWithOptionsTestCase` and `MixedSplitTestCase` by making them simple subclasses of `ModuleStoreTestCase`. Ideally we could eliminate them completely IMHO but that's a project for another day.
* Removes `PureModulestoreTestCase`.

## Testing instructions

Ensure tests are still passing and review code to ensure that what the tests are testing hasn't meaningfully changed.

## Deadline

None
